### PR TITLE
[Models] Add Mistral-7B-Instruct-v0.2 LLM integration

### PR DESCRIPTION
**Pull request by LibbyStarr10**
- MistralForCausalLM: 32 layers, hidden=4096, 32 heads, 8 KV heads
- Standard prefill/decode with StaticCache — zero patches needed
- 7.2B params, sliding window attention, GQA

## Summary

Add Mistral-7B-Instruct-v0.2 as an official model under `models/mistral/`, following the buddy-mlir LLM integration pattern.

**Architecture**: MistralForCausalLM — decoder-only LLM with sliding window attention and grouped-query attention (32 query heads, 8 KV heads). 32 layers, hidden=4096, vocab=32,000.

**Integration approach**: Standard prefill/decode split with `StaticCache`, identical to the Qwen3 pattern. Zero code patches required — the model is a native HuggingFace `MistralForCausalLM` fully compatible with Dynamo.

**Files added** (8 files, 200 lines):
- `models/mistral/CMakeLists.txt` — `buddy_add_model` build config
- `models/mistral/import_model.py` — PyTorch → MLIR importer (prefill + decode)
- `models/mistral/MistralRunner.cpp` — inference runner
- `models/mistral/include/buddy/runtime/models/MistralRunner.h`
- `models/mistral/specs/f32.json` — variant spec (kv_layers=64)
- `models/mistral/validate_accuracy.py` — logits cosine similarity validation
- `models/CMakeLists.txt` — register with `BUDDY_BUILD_MISTRAL_MODEL` guard

## Verification

```bash
cd models/mistral
python import_model.py --output-dir .
ls layer_partitioned/   # forward_prefill.mlir, forward_decode.mlir, subgraph files
ls -lh arg0.data        # ~28 GB (7.2B f32 params)

### CODE REVIEW

**Comment by jiaan-clone** [comment]
Please resolve the code conflicts

**Comment by jiaan-clone** [comment]
Please standardize the code format
